### PR TITLE
【summer_issue#8313】Ecological Construction and optimization of NACOS Service Mesh

### PR DESCRIPTION
**Pull request by RocketEngine26**
## What is the purpose of the change

1. Design Nacos support XDS protocol design document, and complete the module development;

2. Aiming at the performance problem of full push between Nacos and Istio, the incremental push method is used to optimize the design

3. Add the corresponding documentation for the service grid ecology to the NACOS official documentation.

## Brief changelog

1. Change the polling mechanism to listen for event changes;
2. Implement the update design of the data layer、Debounce、CDE, EDS, incremental EDS, incremental MCP.

## Verifying this change

Follow this checklist to help us incorporate your contribution quickly and easily:

* [x] Change the polling mechanism to listen for event changes
* [x] Implement the update design of the data layer
* [x] Debounce
* [x] CDS
* [x] incremental EDS
* [x] incremental MCP

### CODE REVIEW

File: istio/src/main/java/com/alibaba/nacos/istio/common/AbstractConnection.java [hunk 1]
```diff
@@ -27,17 +27,17 @@
  *
  * @author special.fy
  */
-public abstract class AbstractConnection<MessageT> {
+public abstract class AbstractConnection<T> {
```
Comment (SpecialYang) [inline istio/src/main/java/com/alibaba/nacos/istio/common/AbstractConnection.java:30]: Why do you update MessageT as T?

**Review by SpecialYang — CHANGES_REQUESTED** [review]
Please remove irrelative code changes. And remove v2 support and empty generator for CDS/EDS.

File: istio/src/main/java/com/alibaba/nacos/istio/misc/IstioConfig.java [hunk 2]
```diff
@@ -32,8 +32,16 @@ public class IstioConfig {
     private boolean serverEnabled = false;
     @Value("${nacos.istio.mcp.server.port:18848}")
     private int serverPort = 18848;
-    @Value("${nacos.istio.mcp.push.interval:3000}")
-    private int mcpPushInterval;
+    
+    @Value("${nacos.istio.server.full:true}")
+    private boolean fullEnabled = true;
```
Comment (SpecialYang) [inline istio/src/main/java/com/alibaba/nacos/istio/misc/IstioConfig.java:37]: What is the this param does for?

**Review by SpecialYang — CHANGES_REQUESTED** [review]

File: istio/src/main/java/com/alibaba/nacos/istio/misc/IstioConfig.java (hunk 2 above)
Comment (RocketEngine26) [reply istio/src/main/java/com/alibaba/nacos/istio/misc/IstioConfig.java:37]: determine whether to use Delta

File: istio/src/main/java/com/alibaba/nacos/istio/common/NacosServiceInfoResourceWatcher.java [hunk 3]
```diff
@@ -17,85 +17,220 @@
...
-        for (String namespace : namespaces) {
-            Set<Service>  services = ServiceManager.getInstance().getSingletons(namespace);
-            if (services.isEmpty()) {
-                continue;
+    public List<Class<? extends com.alibaba.nacos.common.notify.Event>> subscribeTypes() {
+        List<Class<? extends com.alibaba.nacos.common.notify.Event>> result = new LinkedList<>();
+        //TODO: service data change event, instance event
+        result.add(ClientOperationEvent.ClientRegisterServiceEvent.class);
+        result.add(ClientOperationEvent.ClientDeregisterServiceEvent.class);
+        result.add(InfoChangeEvent.ServiceInfoChangeEvent.class);
```
[the remark this answers is not in the document]
Comment (RocketEngine26) [reply istio/src/main/java/com/alibaba/nacos/istio/common/NacosServiceInfoResourceWatcher.java:82]: The ServiceMetadataEvent is used primarily to determine whether metadata is out of date and can not determine changes to data such as weights

File: istio/src/main/java/com/alibaba/nacos/istio/common/NacosServiceInfoResourceWatcher.java [hunk 4]
```diff
@@ -17,85 +17,183 @@
...
+            Service service = instanceInfoChangeEvent.getService();
+            String serviceName = IstioCrdUtil.buildServiceName(service);
+            
+            boolean full = update(serviceName, service);
+            PushRequest pushRequest = new PushRequest(serviceName, full);
+            pushRequestQueue.add(pushRequest);
+        }
+    }
+    
+    private void init() {
```
Comment (SpecialYang) [inline istio/src/main/java/com/alibaba/nacos/istio/common/NacosServiceInfoResourceWatcher.java:144]: When do we execute init?

File: istio/src/main/java/com/alibaba/nacos/istio/misc/IstioConfig.java (hunk 2 above)
Comment (SpecialYang) [reply istio/src/main/java/com/alibaba/nacos/istio/misc/IstioConfig.java:37]: But this param never be used. consider to remove it.

**Review by SpecialYang — CHANGES_REQUESTED** [review]
Two tips need to be fixed.

File: istio/src/main/java/com/alibaba/nacos/istio/misc/IstioConfig.java (hunk 2 above)
Comment (RocketEngine26) [reply istio/src/main/java/com/alibaba/nacos/istio/misc/IstioConfig.java:37]: This parameter is used in incremental push. If the server does not agree to incremental push even if the client support, incremental push can not be carried out

**Review by SpecialYang — APPROVED** [review]
LGTM

**Pull request merged**